docs/src/references/c8y-log-management.md

```diff
@@ -0,0 +1,122 @@
+whenever it is updated.
+
+Note: If the file `/etc/tedge/c8y/c8y_log_plugin.toml` is ill-formed or cannot be read,
+      then an empty `118` message is sent, indicating no log files are tracked.
+
+## Handling log requests from Cumulocity
+
+This plugin subscribes to `c8y/s/ds` topic, listening for `c8y_LogfileRequest` messages (SmartREST `522`) from Cumulocity, like this one:
+
+```csv
+522,<device-id>,mosquitto,2013-06-22T17:03:14.000+02:00,2013-06-22T18:03:14.000+02:00,ERROR,1000
+```
+
+The plugin then checks the `c8y_log_plugin.toml` file for the log type in the incoming message (`mosquitto`),
+retrives the log file using the `target` glob pattern provided in the plugin config file,
+filters the contents using the date range(`2013-06-22T17:03:14.000+02:00` to `2013-06-22T18:03:14.000+02:00`),
+the search text(`ERROR`) and the maximum line count(`1000`) and uploads this filtered content to Cumulocity.
+
+During this process, Cumulocity is notified of the progress of the `c8y_LogfileRequest` operation
+with SmartREST messages `501`(executing), `502`(failed) or `503`(successful) on `c8y/s/us` topic.
+
+## Updating supported log files from Cumulocity
+
+The supported log files list defined in the `c8y_log_plugin.toml` can be updated both locally as well as from Cumulocity cloud.
+Updates from Cumulocity can be achieved simply by listing this log plugin's confg file(`c8y_log_plugin.toml`) 
+in the configuration file of the `c8y_configuration_plugin`.
+This will enable the `c8y_log_plugin.toml` to be tracked and managed by the `c8y_configuration_plugin`.
+
+## Usage
+
+```shell
+$ c8y_log_plugin
+Plugin supporting log management from Cumulocity on thin-edge.
+USAGE:
+    c8y_log_plugin [OPTIONS]
+OPTIONS:
+        --config-dir <CONFIG_DIR>      [default: /etc/tedge]
+    -h, --help                         Print help information
+    -V, --version                      Print version information
+
+    On start, `c8y_log_plugin` notifies the cloud tenant of the managed log files,
+    listed in the `<CONFIG_DIR>/c8y/c8y_log_plugin.toml`.
+    On receipt of log requests from the cloud, the log files are fetched using the glob path patterns listed in this config file.
+    The cloud tenant is also notified of the progress of the log request operation until success/failure.
+```
+
+## Logging
+
+The `c8y_log_plugin` reports progress and errors to the OS journal which can be retrieved using `journalctl`.
+
+## Future enhancements
+
+To support retieval of logs that are not available as physical files on the file system (journal logs, docker logs, k8s logs etc),
+the `c8y_log_plugin.toml` can be enhanced with additional keys (`journalctl`, `docker`, `kubectl`),
+to list logs that can be retrived with these tools as follows:
+
+```toml
+files = [
+    { type = "mosquitto", target = '/var/log/mosquitto.log' },
+    { type = "software-management", target = '/var/log/tedge/agent/software-management/software-*' },
+]
+
+journalctl = [
```

Comment:
I'm not fully happy with the solution proposed here, because this plugin will have to be extended for each and every kind of log command.

What about providing a command?

```
{ type = "systemd-tedge", command= '/usr/bin/journalctl --since=$FROM --until=$TO --user-unit=tedge' }
```